### DISCUSSION TITLE
Distance Calculation for Traffic Light

### DISCUSSION THREAD

**Discussion by SMRaza1009** [ultralytics/ultralytics#9349] [General]
Dear,

I also have a query. I have traffic light videos. I want to calculate the distance of traffic light signal from a car video. I am currently trying to do with YOLOv8 distance calculator but it seems failed in my case. Because it calculates the distance between the two object. However, I need to calculate distance of traffic light signal from a moving car using coco128 dataset.

Can you assist me this regard? I shall be thankful to you!

**Comment by glenn-jocher** [comment]
@SMRaza1009 hello there! 😊

It sounds like what you're aiming for requires a bit more than just object detection—you might be delving into the realm of monocular depth estimation. Unfortunately, YOLOv8 itself won't directly provide distance measurements from a single camera to an object without additional context or frame of reference, such as comparing two objects' relative sizes or positions which are known a priori.

For your specific scenario of measuring the distance from a car to a traffic light, you could potentially explore integrating depth estimation methods alongside YOLOv8's detection capabilities. One such approach is to use stereo vision (if you have access to a stereo camera setup) or explore single-image depth prediction models that have been trained for similar environments. 

Do bear in mind, the accuracy of your distance estimations would heavily depend on the specifics of your camera setup and the chosen method's suitability to your videos' conditions.

If you're set on using monocular videos (just one camera), consider leveraging additional datasets more focused on depth or distance estimation tasks, which might be more suited for this purpose than coco128.

Wishing you the best in your project, and don't hesitate to reach out if you have more questions!
